src/controllers/shop/add-item.controller.js

```diff
@@ -9,48 +9,81 @@ const addItem = async (req, res) => {
       where: { id },
       attributes: { exclude: ["password"] },
     });
-    const { product_id, qty } = req.body;
+    const products = req.body;
 
-    const product = await Product.findOne({
-      where: { id: product_id },
-      attributes: ["price", "stock", "name_product"],
-    });
-
-    if (qty === 0) {
-      return response(
-        res,
-        400,
-        false,
-        `Minimum quantity for ${product.name_product} is 1`,
-        null
-      );
-    }
+    for (let i = 0; i < products.length; i++) {
+      const { product_id, qty } = products[i];
 
-    const existingCartItem = await Cart.findOne({
-      where: { user_id: user.id, product_id },
-    });
+      const product = await Product.findOne({
+        where: { id: product_id },
+        attributes: ["price", "stock", "name_product"],
+      });
 
-    if (existingCartItem) {
-      if (existingCartItem.qty + qty > product.stock) {
+      if (qty === 0) {
         return response(
           res,
           400,
           false,
-          `Insufficient stock for ${product.name_product}`,
+          `Minimum quantity for ${product.name_product} is 1`,
           null
         );
       }
-      await existingCartItem.update({
-        qty: existingCartItem.qty + qty,
-      });
-    } else {
-      await Cart.create({
-        id: nanoid(10),
-        user_id: user.id,
-        product_id,
-        qty,
-        unit_price: product.price,
+
+      // Add checker if the existing cart item is more than the stock
+      const existingCartItem = await Cart.findOne({
+        where: { user_id: user.id, product_id },
+        include: {
+          model: Transaction,
+          as: "transaction",
+          attributes: ["id"],
+        },
```

Comment:
harus ditambah finding dengan transaction karena kalau kalau misalnya ngeliat dari cart aja dia gak akan bisa ditambah, sehingga kalau semisalnya udah ada cart yang terdaftar di transaction maka akan membuat cart yang baru dengan product yang diinginkan, kasus ini diketahui setelah diuji saat akan menambah item yang sama dengan value yang berbeda, value tidak akan berubah pada cart yang sudah ada.

contohnya seperti dibawah ini: 
![bug-adding-item-to-cart](https://github.com/Belega-Village-UNUD/backend-api-edb/assets/75294452/ee865b0b-9e33-4116-9692-d2e4327056ec)

Reply:
refer to commit: https://github.com/Belega-Village-UNUD/backend-api-edb/pull/51/commits/5dc0312c5a451ca44dfedd2a2349354ef8a3321e